readthedocs/api/v2/utils.py

```diff
@@ -121,6 +121,32 @@ def sync_versions_to_db(project, versions, type):  # pylint: disable=redefined-b
     return added
 
 
+def _create_versions_in_bulk(project, type, versions):
+    """
+    Create versions (tuple of version_id and version_name) in batch.
+
+    Returns the slug of all added versions.
+    """
+    added = set()
+    batch_size = 150
```

Comment:
Where did you get this number? Is 150 a recommended value?

Having 30k versions as you mentioned, is it better to create 150 of them on each iteration or it would be better to create 1k, 5k or all at once?

Reply:
1K is a big number, is slow to evaluate 1k versions on each iteration.